AxonDeepSeg/apply_model.py

```diff
@@ -1,73 +1,166 @@
-        options["overlap_2D"] = [default_overlap, default_overlap]
-
-    # IVADOMED automated segmentation
-    nii_lst, _ = imed_inference.segment_volume(str(path_model), input_filenames, gpu_id=gpu_id, options=options)
-    
-    target_lst = [str(axon_suffix), str(myelin_suffix)]
+    Errors
+    ------
+    ValueError
+        If the class value is not found in the raw prediction.
+    ValueError
+        If the raw nnunet prediction file does not end with '_seg-nnunet.png'.
 
-    imed_inference.pred_to_png(nii_lst, target_lst, str(Path(input_filenames[0]).parent / Path(input_filenames[0]).stem))
-    if verbosity_level >= 1:
-        print(Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(axonmyelin_suffix)))
+    Returns
+    -------
+    new_fname : str
+        Path to the extracted class mask saved.
+    '''
+
+    pred_path = ads_utils.convert_path(pred_path)
+    if not np.any(pred == class_value):
+        raise ValueError(f'Class value {class_value} not found in the raw prediction.')
+    elif not pred_path.name.endswith(str(nnunet_suffix)):
+        raise ValueError(f'Raw nnunet pred file does not end with "{nnunet_suffix}".')
     
-    merge_masks(
-        Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(axon_suffix)),
-        Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(myelin_suffix)), 
-        Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(axonmyelin_suffix))
-        )
+    extraction = np.zeros_like(pred)
+    extraction[pred == class_value] = intensity['binary']
+    new_fname = str(pred_path).replace(str(nnunet_suffix), f'_seg-{class_name}.png')
+    ads_utils.imwrite(new_fname, extraction)
+
+    return new_fname
+
+def axon_segmentation(
+                    path_inputs: List[Path],
+                    path_model: Path,
+                    model_type: Literal['light', 'ensemble']='light',
+                    gpu_id: int=-1,
+                    verbosity_level: int=0,
+                    ) -> NoReturn:
+    '''
+    Segment images by applying a nnU-Net pretrained model.
+
+    Parameters
+    ----------
+    path_inputs : List[pathlib.Path]
+        List of images to segment. We assume they all exist and are already in 
+        the correct format expected by the model (nb of channels, image format).
+    path_model : pathlib.Path
+        Path to the folder of the nnU-Net pretrained model. We assume it exists.
+    model_type : Literal['light', 'ensemble'], optional
+        Type of model, by default 'light'.
+    gpu_id : int, optional
+        GPU ID to use for cuda acceleration. -1 to use CPU, by default -1.
+    verbosity_level : int, optional
+        Level of verbosity, by default 0.
+    '''
+
+    # find all available folds
+    if model_type == 'light':
+        folds_avail = ['all']
+    else:
+        folds_avail = [int(str(f).split('_')[-1]) for f in path_model.glob('fold_*')]
```

Comment:
self: line not covered by tests

Reply:
This line also seems to have a bug; `int(str(f).split('_')[-1])` is trying to convert a string to an int, running this routine just for the light downloaded model, you get a point where `folds_avail` tries to set `int('all')` but throws

```
>   folds_avail = [int(str(f).split('_')[-1]) for f in path_model.glob('fold_*')]
E   ValueError: invalid literal for int() with base 10: 'all'
```

I'm just going to remove the `int()`, @hermancollin let me know if this is wrong and will break something. I'll try testing with the ensemble model structure instead to see if for some reason it would have worked there??

Reply:
Ok so I just downloaded the ensemble generalist model and see the folds there have ints in the name, which is probably why it worked for you.

I'm unsure how I feel about a variable name like this to be a list of strings for one of the if cases, and then a list of ints for the else case. I'll refactor it a bit to be more consistent, let me know @hermancollin if you're not ok with it

Reply:
I think having it always as a string works, because nnunet catches that case and converts it to int here:

https://github.com/MIC-DKFZ/nnUNet/blob/9cd9d80ab3d9542138422f33154ee58421a92088/nnunetv2/inference/predict_from_raw_data.py#L83

Reply:
Done in  69fb7b7

Reply:
Covered by [69fb7b7](https://github.com/axondeepseg/axondeepseg/pull/800/commits/69fb7b75fea3e7cb2dc4a30c5f9f2532e09b0286)

Reply:
everything lgtm here. We will always have either `fold_all` or `fold_#`, as required by the nnunet predictor.